TypeEquality/Teq.fs

```diff
@@ -74,7 +77,8 @@ module Teq =
         /// this assumption is why we don't make this public). Examples of valid values for
         /// f include list, array and option.
         let believeMe<'a, 'b, 'a2, 'b2> (teq : Teq<'a, 'b>) : Teq<'a2, 'b2> =
-            unbox <| (refl : Teq<'a2, 'a2>)
```

Comment:
Good heavens, why did we back-pipe. Fantomas did the right thing, but the source was terrible :P

Reply:
The back-pipe doesn't seem necessary, I remove it.